### DISCUSSION TITLE
`layout={'fill'}` is not as good as `unsized`

### DISCUSSION THREAD

**Discussion by jahirfiquitiva** [vercel/next.js#19687] [Ideas]
# Feature request

## Is your feature request related to a problem? Please describe.

When I first started using the new `Image` component, the `unsized` prop was good for external images which dimensions I didn't know but still wanted to be optimized.

After `layout={'fill'}` it basically forces me to set a width and height, not for the image, but for the `div` that contains it, which makes me see no difference between having to set a `width` and `height` for the `Image` component.

The problem is, there are some images I just want to display with full width, and auto height for them to keep the aspect ratio, and I don't want to think about its specific width or height. Now, I have to wrap them in a div with a specific width and height, just as you did in [your examples](https://github.com/vercel/next.js/blob/canary/examples/image-component/pages/layout-fill.js).

## Describe the solution you'd like

I would like to have an optimization option that really doesn't force me to set a specific width and height for the `Image` component neither the parent `div`.

## Additional context

Here I show you two screenshots of an the same image

### Image using next `10.0.1` with `unsized` 

<details><summary><b>Code used</b></summary>

```javascript
  <Image
    unsized
    loading={'lazy'}
    src={'/assets/images/gifs/mail.gif'}
    alt={'Dog checking mail'}/>
```

</details/>

<img width="512" alt="Shot 2020-11-30 at 13-12-44@2x" src="https://user-images.githubusercontent.com/10360816/100649621-b25f1080-3310-11eb-8e7e-bd82c0e65c42.png">

### Image using next `10.0.3` with `layout={'fill'}`

<details><summary><b>Code used</b></summary>

```javascript
  <Image
    layout={'fill'}
    loading={'lazy'}
    src={'/assets/images/gifs/mail.gif'}
    alt={'Dog checking mail'}/>
```

</details/>

<img width="512" alt="Shot 2020-11-30 at 13-12-53@2x" src="https://user-images.githubusercontent.com/10360816/100649701-d6225680-3310-11eb-8026-6edb8b6b85f3.png">

### Image using next `10.0.3` with `layout={'fill'}` and `objectFit={'contain'}`

<details><summary><b>Code used</b></summary>

```javascript
  <Image
    layout={'fill'}
    objectFit={'contain'}
    loading={'lazy'}
    src={'/assets/images/gifs/mail.gif'}
    alt={'Dog checking mail'}/>
```

</details/>

<img width="512" alt="Shot 2020-11-30 at 13-12-53@2x" src="https://user-images.githubusercontent.com/10360816/100650344-c9523280-3311-11eb-9dd1-c6f3c822e74e.png">

**Comment by ryanovas** [comment]
I would also like to see this basic functionality. It's absolutely crazy to me that there is no easy way to simply do width: 100% and height: auto. I can't remember a single instance since I was first learning HTML that I had to manually set a width and height on an image. There is layout: fill but that is not the same. That means I still have to set a width and height on a parent container. I can't simply let the image take up space with it's native aspect ratio. It's the difference between setting background-image on a div with CSS vs putting an actual img tag, and there's a million reasons why those must be separate.
